lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -172,7 +177,14 @@ class BitmapCanvas extends EngineCanvas {
     _canvasPool.clear();
     final int len = _children.length;
     for (int i = 0; i < len; i++) {
-      _children[i].remove();
+      html.Element child = _children[i];
+      if (child is html.ImageElement) {
+        _reusableImages ??= {};
+        final String src = child.src;
+        _reusableImages[src] ??= []..add(child);
```

Comment:
Instead of adding the image during deconstruction of the canvas, I would add the image at the time it is painted. This way we don't have to read back from the DOM, and we can still reuse the image even if the canvas is not reused but a new one is painted.

Reply:
Done.